apps/voting/app/src/App.js

```diff
@@ -212,60 +222,67 @@ class App extends React.Component {
         ? null
         : preparedVotes.find(vote => vote.voteId === currentVoteId)
     const hasCurrentVote = appStateReady && Boolean(currentVote)
-
     return (
-      <div css="min-width: 320px">
-        <Main assetsUrl="./aragon-ui">
-          <AppLayout
-            title="Voting"
-            onMenuOpen={this.handleMenuPanelOpen}
-            mainButton={{
-              label: 'New vote',
-              icon: <NewVoteIcon />,
-              onClick: this.handleCreateVoteOpen,
-            }}
-          >
-            {appStateReady && votes.length > 0 ? (
-              <Votes votes={preparedVotes} onSelectVote={this.handleVoteOpen} />
-            ) : (
-              <EmptyState onActivate={this.handleCreateVoteOpen} />
-            )}
-          </AppLayout>
-          <SidePanel
-            title={`Vote #${currentVoteId} (${
-              currentVote && currentVote.data.open ? 'Open' : 'Closed'
-            })`}
-            opened={hasCurrentVote && !createVoteVisible && voteVisible}
-            onClose={this.handleVoteClose}
-            onTransitionEnd={this.handleVoteTransitionEnd}
-          >
-            {hasCurrentVote && (
-              <VotePanelContent
-                app={app}
-                vote={currentVote}
-                user={userAccount}
-                ready={voteSidebarOpened}
-                tokenContract={tokenContract}
-                tokenDecimals={tokenDecimals}
-                tokenSymbol={tokenSymbol}
-                onVote={this.handleVote}
-                onExecute={this.handleExecute}
-              />
-            )}
-          </SidePanel>
+      <IdentityProvider
```

Comment:
Wondering how you feel about this @bpierre; I think I'd generally recommend putting `<Main />` as the very first component that's rendered.

Reply:
Yes let’s do this 👍

The only one that really need to be outside is the `<div css="min-width: 320px">`, so it applies to the elements rendered at the root level.

Reply:
Two questions:
- Is `css` a valid prop? Why is it used over `style`
- Can we add `min-width: 320px` to the Main element?

Reply:
> Is css a valid prop? Why is it used over style

It is being transformed by the styled-components [babel plugin](https://www.styled-components.com/docs/api#css-prop).

It transforms this:

```jsx
import React from 'react'

const App = () => <div css="min-width: 320px" />
```

Into this:

```jsx
import React from 'react'
import styled from 'styled-components'

const App = () => <A />

const A = styled.div`
  min-width: 320px
`
```

And can it even works if you do this:


```jsx
import React from 'react'

const items = ['apple', 'orange', 'banana']

const App = () => (
  <ul>
    {items.map(item => (
      // Only one styled component will be extracted by the babel transform.
      <li css="color: tomato">{item}</li>
    ))}
  </ul>
)
```

It is basically `style`, but without the downsides :-)

> Can we add `min-width: 320px` to the `Main` element?

That’s something I keep wondering… We used to have `AragonApp`, which was defining some default styles, and it was causing too many issues. `Main` replaced it, and it was decided that  it would not set anything styles, and only providing context. We can imagine a website widget using aragonUI, and being smaller than `320px`, so it would be a problem to have it by default.

But on the other hand, developing a fullscreen app with aragonUI is quite a common use case… maybe we could have it as a prop, like `minWidth`, that would be either `0` or `320` by default?

Reply:
Thanks for the clarification.

> Can we add min-width: 320px to the Main element?

On that note, I actually meant to have it inline: `<Main assetsUrl="./aragon-ui" css="min-width: 320px">` rather than a separate element. 

> But on the other hand, developing a fullscreen app with aragonUI is quite a common use case… maybe we could have it as a prop, like minWidth, that would be either 0 or 320 by default?

I don't have an opinion on that.

Reply:
Oh I see, no it wouldn’t directly work because `Main` doesn’t render any DOM element. We could redirect styling props (`style` and `className`) to `Root`, but I would like to not encourage doing this, and prefer the `minWidth` solution for now.

@sohkai @AquiGorka any thoughts on that?

Reply:
`html, body { min-width: 320px}`?